frontend/src/components/student/studentList.tsx

```diff
@@ -60,13 +62,13 @@ export const StudentList = (props: { isDraggable: boolean }) => {
                                 event.dataTransfer.setData("url", student._links.self.href);
                                 event.dataTransfer.setData("name", student.firstName);
                             }}
-                            // Should be changed to individual student page later
-                            onClick={() => {
+                            onClick={async () => {
                                 if (!draggable) {
-                                    let studentPath: string = extractIdFromStudentUrl(
-                                        student._links.self.href
-                                    );
-                                    router.push(transformer("/" + studentPath)).catch(console.log);
+                                    let studentId: string = extractIdFromStudentUrl(student._links.self.href);
+                                    await router.replace({
+                                        pathname: "/" + applicationPaths.students + "/" + studentId,
+                                        query: { ...router.query },
```

Comment:
This is to make sure the query is kept when selecting a student right?

Reply:
Yes indeed